core/src/main/java/org/apache/calcite/sql/SqlUnnestOperator.java

```diff
@@ -97,6 +97,17 @@ public SqlUnnestOperator(boolean withOrdinality) {
     return builder.build();
   }
 
+  private boolean allowFlattenStruct(SqlOperatorBinding operatorBinding) {
+    if (!(operatorBinding instanceof SqlCallBinding)) {
```

Comment:
`allowFlattenStruct` -> `allowAliasUnnestArrayColumns` ?

Reply:
Another class type for operator binding is `RexCallBinding`, but why we return true for that ?

Reply:
@danny0405 Took a second thought on the function name, the result of the function is actually whether **disallow flatten struct in UNNEST**. So maybe disallowFlattenStructInUnnest? Let me know. Thanks.

Reply:
I mean `RexCallBinding` comes from building the call from `RelBuilder`, but why we return true for that, confusing for me.

Reply:
Again, this is original behavior. I feel like for a PR, it could be better to only contain all the relevant changes without changing existing behaviors for irrelevant flows. 

There are 8 other subclasses of `SqlOperatorBinding` and I guess several of them should not be here. If you believe this is a serious issue and should be addressed, we should have another refactor for it.

Let me know if you insist on gating the `RexCallBinding`, I will add another commit to do it.

Reply:
I know you only to support `SqlCallBinding` for unnest alias columns, but the logic seems weird, can you change
`allowFlattenStruct` => `allowAliasUnnestColumns` ?

Reply:
I think it looks even more confusing. If you read the code carefully enough, it is **NOT** allowAliasUnnestColumns. If you prefer the name change, I think `disallowAliasUnnestColumn` is more appropriate.

Reply:
Couldn’t you use !allowAliasUnnestColumns ?

Reply:
That's a good idea.

Reply:
Good idea. Will do